### PR TITLE
Use sync2.Cycle in agreement sender

### PR DESCRIPTION
**Pull request by mniewrzal**
This change replaces `Ticker` with `sync2.Cycle` and introduce basic tests for sending agreements to satellite.

### CODE REVIEW

File: pkg/piecestore/psserver/agreementsender/agreementsender_test.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+			assert.Equal(t, 1, len(allocPerSat))
+			putAllocation[node.ID()] = allocPerSat[0].Agreement.Total
+		}
+	}
+
+	for _, node := range planet.StorageNodes {
+		node.Agreements.Sender.Loop.Trigger()
+	}
+
+	time.Sleep(1 * time.Second)
```
Comment (egonelbre) [inline pkg/piecestore/psserver/agreementsender/agreementsender_test.go]: Is this sleep necessary? There are quite a few methods on cycle that allows for it to be completed.

File: pkg/piecestore/psserver/agreementsender/agreementsender_test.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+			// sum with previous upload PUTs
+			putAllocation[node.ID()] += allocPerSat[0].Agreement.Total
+		}
+	}
+
+	for _, node := range planet.StorageNodes {
+		node.Agreements.Sender.Loop.Trigger()
+	}
+
+	time.Sleep(1 * time.Second)
```
Comment (egonelbre) [inline pkg/piecestore/psserver/agreementsender/agreementsender_test.go]: Same here, probably not necessary.

File: pkg/piecestore/psserver/agreementsender/agreementsender_test.go (hunk 1 above)
Comment (mniewrzal) [reply pkg/piecestore/psserver/agreementsender/agreementsender_test.go]: it fails locally without Sleep so most probably it will fail on travis

Comment (egonelbre) [reply pkg/piecestore/psserver/agreementsender/agreementsender_test.go]: Use `TriggerWait`.

Comment (mniewrzal) [reply pkg/piecestore/psserver/agreementsender/agreementsender_test.go]: ah I misunderstand earlier whats `TriggerWait` is doing. Now its working without sleep. Thanks

**Comment by mniewrzal** [comment]
wont be applicable to new piecestore - #1422

**Pull request closed without merging**